controller/src/beerocks/bml/internal/bml_internal.cpp

```diff
@@ -1767,6 +1839,176 @@ int bml_internal::start_dcs_single_scan(const sMacAddr &mac, int dwell_time_ms,
+            cmdu_tx);
+
+    if (!request) {
+        LOG(ERROR) << "Failed building cACTION_BML_CLIENT_SET_CLIENT_REQUEST message!";
+        return (-BML_RET_OP_FAILED);
+    }
+
+    request->sta_mac()                               = sta_mac;
+    request->client_config().stay_on_initial_radio   = client_config.stay_on_initial_radio;
+    request->client_config().stay_on_selected_device = client_config.stay_on_selected_device;
+    request->client_config().selected_bands =
+        beerocks_message::eClientSelectedBands(client_config.selected_bands);
+
+    int result = 0;
+    if (send_bml_cmdu(result, request->get_action_op()) != BML_RET_OK) {
+        LOG(ERROR) << "Send cACTION_BML_CLIENT_SET_CLIENT_REQUEST failed";
+        return (-BML_RET_OP_FAILED);
+    }
+
+    if (result != 0) {
+        LOG(ERROR) << "cACTION_BML_CLIENT_SET_CLIENT_REQUEST returned error code:" << result;
+        return result;
+    }
+
+    return BML_RET_OK;
+}
+
+int bml_internal::client_get_client(const sMacAddr &sta_mac, BML_CLIENT &client)
+{
+    LOG(DEBUG) << "client_get_client";
+
+    // If the socket is not valid, attempt to re-establish the connection
+    if (!m_sockMaster) {
+        int iRet = connect_to_master();
+        if (iRet != BML_RET_OK) {
+            return iRet;
+        }
+    }
+
+    // Initialize the promise for receiving the response
+    beerocks::promise<bool> prmClientGet;
+    m_prmClientGet = &prmClientGet;
+    int iOpTimeout = RESPONSE_TIMEOUT; // Default timeout
+    m_client       = &client;
+
+    auto request =
+        message_com::create_vs_message<beerocks_message::cACTION_BML_CLIENT_GET_CLIENT_REQUEST>(
+            cmdu_tx);
+
+    if (!request) {
+        LOG(ERROR) << "Failed building cACTION_BML_CLIENT_GET_CLIENT_REQUEST message!";
+        return (-BML_RET_OP_FAILED);
+    }
+
+    request->sta_mac() = sta_mac;
+    // Build and send the message
+    if (!message_com::send_cmdu(m_sockMaster, cmdu_tx)) {
+        LOG(ERROR) << "Failed sending param get message!";
+        m_prmClientGet = nullptr;
+        m_client       = nullptr;
+        return (-BML_RET_OP_FAILED);
+    }
+    LOG(DEBUG) << "cACTION_BML_CLIENT_GET_CLIENT_REQUEST sent";
+
+    int iRet = BML_RET_OK;
+
+    if (!m_prmClientGet->wait_for(iOpTimeout)) {
+        LOG(WARNING) << "Timeout while waiting for client get response...";
+        iRet = -BML_RET_TIMEOUT;
+    }
+
+    // Clear the scan results members
+    m_client = nullptr;
+
+    // Clear the promise holder
+    m_prmClientGet = nullptr;
+
+    if (iRet != BML_RET_OK) {
+        LOG(ERROR) << "Results get failed!";
+        return (iRet);
+    }
+
+    if (iRet != 0) {
+        LOG(ERROR) << "Client list returned with error code:" << iRet << ". Aborting!";
+        return iRet;
+    }
+
+    return BML_RET_OK;
+}
+
+int bml_internal::client_dump_client(const sMacAddr &sta_mac)
+{
+    LOG(DEBUG) << "client_dump_client";
```

Comment:
you are missing the implementation of the `client_dump_client`

Reply:
Removed the `client_dump_client` since the `client_get_client` prints the selected client in the BML CLI. so I don't see the need for a `dump_client` flow.